pkg/test/runner/runner.go

```diff
@@ -148,6 +148,12 @@ func (r *Runner) runPipeline() error {
 		fnErr = command.Execute()
 	}
 
+	// check for errors, because compareResult doesn't have access to the
+	// command's exit code for pipeline
+	if fnErr != nil && r.testCase.Config.ExitCode != 0 {
```

Comment:
If the pipeline is expected to fail, we should not keep looping it in line 146.

`compareResult` can get the exit code from `fnErr`.

Reply:
I changed it above so that it breaks when fnErr != nil.

For `compareResult`, it cannot do this for pipeline because I am running it with `cobra.Command.Execute` which does not return `exec.ExitError` as expected by `compareResult` (`fn run` test uses `exec.Command`). I could change pipeline to use `exec.Command` but that would require the developer to rebuild kpt locally every time they make changes, which doesn't seem ideal to me and could become tedious

Reply:
If pipeline fails, the error message is printed in the terminal.